### PR TITLE
Fixed #24788 - Allow Forms to specify a prefix at the class level

### PR DESCRIPTION
**Pull request by pwmarcz**
https://code.djangoproject.com/ticket/24788

### CODE REVIEW

File: tests/forms_tests/tests/test_forms.py [hunk 1]
```diff
@@ -1671,6 +1671,20 @@ def add_prefix(self, field_name):
         self.assertEqual(p.cleaned_data['last_name'], 'Lennon')
         self.assertEqual(p.cleaned_data['birthday'], datetime.date(1940, 10, 9))
 
+        # Prefix can be also specified at the class level.
```
Comment (timgraham) [inline tests/forms_tests/tests/test_forms.py]: `def test_class_prefix(self):`

**Comment by timgraham** [comment]
Doc updates/release notes?

**Comment by pwmarcz** [comment]
Done.

File: docs/ref/forms/api.txt [hunk 2]
```diff
@@ -1065,3 +1065,15 @@ You can put several Django forms inside one ``<form>`` tag. To give each
     >>> print(father.as_ul())
     <li><label for="id_father-first_name">First name:</label> <input type="text" name="father-first_name" id="id_father-first_name" /></li>
     <li><label for="id_father-last_name">Last name:</label> <input type="text" name="father-last_name" id="id_father-last_name" /></li>
+
+The prefix can also be specified at the class level::
```
Comment (timgraham) [inline docs/ref/forms/api.txt]: need a `versionchanged:: 1.9` annotation

File: docs/ref/forms/api.txt [hunk 3]
```diff
@@ -1065,3 +1065,15 @@ You can put several Django forms inside one ``<form>`` tag. To give each
...
+
+    >>> class MotherForm(forms.Form):
+    ...     prefix = 'mother'
+    ...     first_name = forms.CharField()
```
Comment (timgraham) [inline docs/ref/forms/api.txt]: I think the fields/example could be omitted as the functionality is the same as when specified at the instance level.

File: tests/forms_tests/tests/test_forms.py [hunk 4]
```diff
@@ -1671,6 +1671,21 @@ def add_prefix(self, field_name):
         self.assertEqual(p.cleaned_data['last_name'], 'Lennon')
         self.assertEqual(p.cleaned_data['birthday'], datetime.date(1940, 10, 9))
 
+    def test_class_prefix(self):
+        # Prefix can be also specified at the class level.
+        class Person(Form):
+            prefix = 'foo'
+
+            first_name = CharField()
```
Comment (timgraham) [inline tests/forms_tests/tests/test_forms.py]: only need 1 field really, right? I'd tend to always put the fields first when defining the class.

**Comment by pwmarcz** [comment]
You mean remove the example entirely?

How about now?

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
merged in 4df7e8483b2679fc1cba3410f08960bac6f51115, thanks!

**Pull request closed without merging**